### PR TITLE
Feature/feature crud groups

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ communitymanager/src/main/java/org/pesmypetcare/communitymanager/managers/GroupManagerClient.java  100
+ usermanager/src/main/java/org/pesmypetcare/usermanager/clients/user/UserManagerClient.java  8
         

Complexity increasing per file
==============================
- communitymanager/src/main/java/org/pesmypetcare/communitymanager/datacontainers/TagData.java  3
- communitymanager/src/test/java/org/pesmypetcare/communitymanager/managers/GroupManagerClientTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434&bid=18196181)

File: app/build.gradle [hunk 1]
```diff
@@ -44,9 +46,9 @@ dependencies {
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation "androidx.lifecycle:lifecycle-viewmodel:2.2.0"
     implementation "androidx.lifecycle:lifecycle-livedata:2.2.0"
-    implementation project(path: ':usermanager')
-    implementation project(path: ':communitymanager')
-    implementation project(path: ':httptools')
+    //implementation project(path: ':usermanager')
+    //implementation project(path: ':communitymanager')
+    //implementation project(path: ':httptools')
```
Comment (marclll) [inline app/build.gradle]: Why commented?

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- communitymanager/src/main/java/org/pesmypetcare/communitymanager/datacontainers/TagData.java  3
- communitymanager/src/test/java/org/pesmypetcare/communitymanager/managers/GroupManagerClientTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434&bid=18196181)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+                + "          \"latitude\":84,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
+                + "          \"latitude\":34,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
+                + "          \"latitude\":21,\n"
+                + "          \"longitude\":58\n"
+                + "        }\n"
+                + "      ],\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:117]: Codacy Issue found: [The String "      ],\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 51
           

Complexity increasing per file
==============================
- communitymanager/src/main/java/org/pesmypetcare/communitymanager/datacontainers/TagData.java  3
- communitymanager/src/test/java/org/pesmypetcare/communitymanager/managers/GroupManagerClientTest.java  1
- usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434&bid=18196181)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+                + "  },\n"
+                + "  \"key\": \"1998-01-08T15:20:30\"\n"
+                + "}\n"
+                + "]");
+        washDataJson = new StringBuilder("{\n"
+                + "    \"description\": \"It smelt so bad\",\n"
+                + "    \"duration\": 54\n"
+                + "}"
+        );
+        washData = new WashData("It smelt so bad", 54);
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:384]: Codacy Issue found: ['54' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+                + "        {\n"
+                + "          \"latitude\":84,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
+                + "          \"latitude\":34,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
+                + "          \"latitude\":21,\n"
+                + "          \"longitude\":58\n"
+                + "        }\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:116]: Codacy Issue found: [The String "        }\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 5]
```diff
@@ -0,0 +1,644 @@
...
+                + "      ],\n"
+                + "      \"description\":\"Staying horizontal over a plain surface\",\n"
+                + "      \"endDateTime\":\"2000-01-08T15:20:30\"\n"
+                + "    },\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:120]: Codacy Issue found: [The String "    },\n" appears 22 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 6]
```diff
@@ -0,0 +1,644 @@
...
+                + "        },{\n"
+                + "          \"latitude\":34,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
+                + "          \"latitude\":21,\n"
+                + "          \"longitude\":58\n"
+                + "        }\n"
+                + "      ],\n"
+                + "      \"description\":\"Staying horizontal over a plain surface\",\n"
+                + "      \"endDateTime\":\"2000-01-08T15:20:30\"\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:119]: Codacy Issue found: [The String "      \"endDateTime\":\"2000-01-08T15:20:30\"\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 7]
```diff
@@ -0,0 +1,644 @@
...
+        washData = new WashData("It smelt so bad", 54);
+        washCollectionList = new ArrayList<>();
+        washCollectionList.add(new Wash(DATE_1, washData));
+        washCollectionList.add(new Wash(DATE_2, washData));
+        washCollectionList.add(new Wash(DATE_3, washData));
+
+        weightCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\": {\n"
+                + "      \"value\": 54\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:393]: Codacy Issue found: [The String "      \"value\": 54\n" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 8]
```diff
@@ -0,0 +1,644 @@
...
+    @Before
+    public void setUp() {
+        // Corregir exercise per ficar coordinates
+        exerciseCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\":{\n"
+                + "      \"name\":\"Planking\",\n"
+                + "      \"coordinates\":[\n"
+                + "        {\n"
+                + "          \"latitude\":84,\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:108]: Codacy Issue found: [The String "          \"latitude\":84,\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 9]
```diff
@@ -0,0 +1,644 @@
...
+                + "          \"longitude\":58\n"
+                + "        }\n"
+                + "      ],\n"
+                + "      \"description\":\"Staying horizontal over a plain surface\",\n"
+                + "      \"endDateTime\":\"2000-01-08T15:20:30\"\n"
+                + "    },\n"
+                + "    \"key\":\"1998-01-08T15:20:30\"\n"
+                + "  }\n"
+                + "]");
+        exerciseDataJson = new StringBuilder("{\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:160]: Codacy Issue found: [The String "{\n" appears 8 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 10]
```diff
@@ -0,0 +1,644 @@
...
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Before
+    public void setUp() {
+        // Corregir exercise per ficar coordinates
+        exerciseCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\":{\n"
+                + "      \"name\":\"Planking\",\n"
+                + "      \"coordinates\":[\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:106]: Codacy Issue found: [The String "      \"coordinates\":[\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 11]
```diff
@@ -0,0 +1,644 @@
...
+        exerciseData = new ExerciseData("Planking", "Staying horizontal over a plain surface",
+                "2000-01-08T15:20:30", coordinates);
+        exerciseCollectionList = new ArrayList<>();
+        exerciseCollectionList.add(new Exercise(DATE_1, exerciseData));
+        exerciseCollectionList.add(new Exercise(DATE_2, exerciseData));
+        exerciseCollectionList.add(new Exercise(DATE_3, exerciseData));
+
+        illnessCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\": {\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:191]: Codacy Issue found: [The String "    \"body\": {\n" appears 19 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 12]
```diff
@@ -0,0 +1,644 @@
...
+    private PetCollectionsManagerClient client = new PetCollectionsManagerClient();
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Before
+    public void setUp() {
+        // Corregir exercise per ficar coordinates
+        exerciseCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:103]: Codacy Issue found: [The String "  {\n" appears 8 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 13]
```diff
@@ -0,0 +1,644 @@
...
+                + "          \"longitude\":58\n"
+                + "        }\n"
+                + "      ],\n"
+                + "      \"description\":\"Staying horizontal over a plain surface\",\n"
+                + "      \"endDateTime\":\"2000-01-08T15:20:30\"\n"
+                + "}"
+        );
+        List<LatLng> coordinates = new ArrayList<>();
+        coordinates.add(new LatLng(84, 58));
+        coordinates.add(new LatLng(34, 58));
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:180]: Codacy Issue found: ['34' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 14]
```diff
@@ -0,0 +1,644 @@
...
+                + "      \"value\": 54\n"
+                + "    },\n"
+                + "    \"key\": \"1998-01-08T15:20:30\"\n"
+                + "  }\n"
+                + "]");
+        weightDataJson = new StringBuilder("{\n"
+                + " \"value\": 54\n"
+                + "}"
+        );
+        weightData = new WeightData(54);
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:412]: Codacy Issue found: ['54' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 15]
```diff
@@ -0,0 +1,644 @@
...
+    private TaskManager taskManager;
+
+    @InjectMocks
+    private PetCollectionsManagerClient client = new PetCollectionsManagerClient();
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Before
+    public void setUp() {
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:100]: Codacy Issue found: [Avoid really long methods.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 16]
```diff
@@ -0,0 +1,644 @@
...
+        vaccinationData = new VaccinationData("It was really pleasant");
+        vaccinationCollectionList = new ArrayList<>();
+        vaccinationCollectionList.add(new Vaccination(DATE_1, vaccinationData));
+        vaccinationCollectionList.add(new Vaccination(DATE_2, vaccinationData));
+        vaccinationCollectionList.add(new Vaccination(DATE_3, vaccinationData));
+
+        vetVisitCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\": {\n"
+                + "      \"reason\": \"It was really pleasant\",\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:329]: Codacy Issue found: [The String "      \"reason\": \"It was really pleasant\",\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 17]
```diff
@@ -0,0 +1,644 @@
...
+    @InjectMocks
+    private PetCollectionsManagerClient client = new PetCollectionsManagerClient();
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Before
+    public void setUp() {
+        // Corregir exercise per ficar coordinates
+        exerciseCollectionJson = new StringBuilder("[\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:102]: Codacy Issue found: [The String "[\n" appears 8 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 18]
```diff
@@ -0,0 +1,644 @@
...
+        washCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\": {\n"
+                + "      \"description\": \"It smelt so bad\",\n"
+                + "      \"duration\": 54\n"
+                + "    },\n"
+                + "    \"key\": \"1990-01-08T15:20:30\"\n"
+                + "  },{\n"
+                + "  \"body\": {\n"
+                + "    \"description\": \"It smelt so bad\",\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:367]: Codacy Issue found: [The String "    \"description\": \"It smelt so bad\",\n" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 19]
```diff
@@ -0,0 +1,644 @@
...
+        // Corregir exercise per ficar coordinates
+        exerciseCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\":{\n"
+                + "      \"name\":\"Planking\",\n"
+                + "      \"coordinates\":[\n"
+                + "        {\n"
+                + "          \"latitude\":84,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:110]: Codacy Issue found: [The String "        },{\n" appears 8 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 20]
```diff
@@ -0,0 +1,644 @@
...
+                + "    },\n"
+                + "    \"key\": \"1990-01-08T15:20:30\"\n"
+                + "  },{\n"
+                + "    \"body\": {\n"
+                + "      \"endDateTime\": \"2000-01-08T15:20:30\",\n"
+                + "      \"description\": \"I'm not feeling so well\",\n"
+                + "      \"type\": \"Normal\",\n"
+                + "      \"severity\": \"High\"\n"
+                + "    },\n"
+                + "    \"key\": \"1995-01-08T15:20:30\"\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:205]: Codacy Issue found: [The String "    \"key\": \"1995-01-08T15:20:30\"\n" appears 5 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 21]
```diff
@@ -0,0 +1,644 @@
...
+                + "      \"description\":\"Staying horizontal over a plain surface\",\n"
+                + "      \"endDateTime\":\"2000-01-08T15:20:30\"\n"
+                + "}"
+        );
+        List<LatLng> coordinates = new ArrayList<>();
+        coordinates.add(new LatLng(84, 58));
+        coordinates.add(new LatLng(34, 58));
+        coordinates.add(new LatLng(21, 58));
+        exerciseData = new ExerciseData("Planking", "Staying horizontal over a plain surface",
+                "2000-01-08T15:20:30", coordinates);
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:183]: Codacy Issue found: [The String "2000-01-08T15:20:30" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 22]
```diff
@@ -0,0 +1,644 @@
...
+                + "        },{\n"
+                + "          \"latitude\":21,\n"
+                + "          \"longitude\":58\n"
+                + "        }\n"
+                + "      ],\n"
+                + "      \"description\":\"Staying horizontal over a plain surface\",\n"
+                + "      \"endDateTime\":\"2000-01-08T15:20:30\"\n"
+                + "    },\n"
+                + "    \"key\":\"1990-01-08T15:20:30\"\n"
+                + "  },{\"body\":{\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:122]: Codacy Issue found: [The String "  },{\"body\":{\n" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 23]
```diff
@@ -0,0 +1,644 @@
...
+        exerciseCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\":{\n"
+                + "      \"name\":\"Planking\",\n"
+                + "      \"coordinates\":[\n"
+                + "        {\n"
+                + "          \"latitude\":84,\n"
+                + "          \"longitude\":58\n"
+                + "        },{\n"
+                + "          \"latitude\":34,\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:111]: Codacy Issue found: [The String "          \"latitude\":34,\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 24]
```diff
@@ -0,0 +1,644 @@
...
+                + "  {\n"
+                + "    \"body\": {\n"
+                + "      \"description\": \"It smelt so bad\",\n"
+                + "      \"duration\": 54\n"
+                + "    },\n"
+                + "    \"key\": \"1990-01-08T15:20:30\"\n"
+                + "  },{\n"
+                + "  \"body\": {\n"
+                + "    \"description\": \"It smelt so bad\",\n"
+                + "    \"duration\": 54\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:368]: Codacy Issue found: [The String "    \"duration\": 54\n" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 25]
```diff
@@ -0,0 +1,644 @@
...
+        mealData = new MealData("Tortilla", 85.44);
+        mealCollectionList = new ArrayList<>();
+        mealCollectionList.add(new Meal(DATE_1, mealData));
+        mealCollectionList.add(new Meal(DATE_2, mealData));
+        mealCollectionList.add(new Meal(DATE_3, mealData));
+
+        medicationCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\": {\n"
+                + "      \"quantity\": 85.0,\n"
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:265]: Codacy Issue found: [The String "      \"quantity\": 85.0,\n" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ usermanager/src/main/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClient.java  91
+ communitymanager/src/main/java/org/pesmypetcare/communitymanager/managers/GroupManagerClient.java  100
+ usermanager/src/main/java/org/pesmypetcare/usermanager/clients/user/UserManagerClient.java  8
         

Complexity increasing per file
==============================
- communitymanager/src/main/java/org/pesmypetcare/communitymanager/datacontainers/TagData.java  3
- usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java  1
- communitymanager/src/test/java/org/pesmypetcare/communitymanager/managers/GroupManagerClientTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434&bid=18196181)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 26]
```diff
@@ -0,0 +1,644 @@
...
+        assertEquals("Should return a list", exerciseCollectionList, response);
+    }
+
+    @Test
+    public void getExercises() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(exerciseDataJson);
+        ExerciseData response = client.getExercise(ACCESS_TOKEN, USERNAME, PET_NAME, DATE_1);
+        assertEquals("Should return the specified element", exerciseData, response);
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:443]: Codacy Issue found: [The String "Should return the specified element" appears 9 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

File: usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java [hunk 27]
```diff
@@ -0,0 +1,644 @@
...
+        weightCollectionList.add(new Weight(DATE_3, weightData));
+    }
+
+    @Test
+    public void getAllExercises() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(exerciseCollectionJson);
+        List<Exercise> response = client.getAllExercises(ACCESS_TOKEN, USERNAME, PET_NAME);
+        assertEquals("Should return a list", exerciseCollectionList, response);
```
Comment (santidrj) [inline usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java:425]: Codacy Issue found: [The String "Should return a list" appears 16 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ usermanager/src/main/java/org/pesmypetcare/usermanager/clients/user/UserManagerClient.java  8
+ usermanager/src/main/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClient.java  91
+ communitymanager/src/main/java/org/pesmypetcare/communitymanager/managers/GroupManagerClient.java  100
         

Complexity increasing per file
==============================
- communitymanager/src/test/java/org/pesmypetcare/communitymanager/managers/GroupManagerClientTest.java  1
- usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java  1
- communitymanager/src/main/java/org/pesmypetcare/communitymanager/datacontainers/TagData.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434&bid=18196181)

**Pull request merged**